### PR TITLE
chore: mock paths conflict detect

### PR DESCRIPTION
**Pull request by ycjcl868**
image

##### Checklist

- [x] `npm test` passes
- [x] tests are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- any feature?
- close https://github.com/umijs/umi/ISSUE_URL

### CODE REVIEW

File: packages/preset-built-in/src/plugins/commands/dev/mock/createMiddleware.ts [hunk 1]
```diff
@@ -25,12 +25,12 @@ export default function (opts = {} as IMockOpts): ICreateMiddleware {
...
   watcher
     .on('ready', () => debug('Initial scan complete. Ready for changes'))
-    .on('all', (event, file) => {
+    .on('all', async (event, file) => {
       debug(`[${event}] ${file}, reload mock data`);
       errors.splice(0, errors.length);
       cleanRequireCache(mockWatcherPaths);
       // refresh data
-      data = updateMockData()?.mockData;
+      data = (await updateMockData())?.mockData;
```
Comment (ycjcl868) [inline packages/preset-built-in/src/plugins/commands/dev/mock/createMiddleware.ts:33]: `async` 也是可以 run 的

Comment (ycjcl868) [reply packages/preset-built-in/src/plugins/commands/dev/mock/createMiddleware.ts:33]: https://stackoverflow.com/questions/47448567/using-async-in-event-emitter

File: packages/preset-built-in/src/plugins/commands/dev/mock/utils.test.ts [hunk 2]
```diff
@@ -213,3 +214,64 @@ describe('umi-mock:getMockData', () => {
     });
   });
 });
+
+test('getConflictPaths', async () => {
+  const routes = [
```
Comment (sorrycc) [inline packages/preset-built-in/src/plugins/commands/dev/mock/utils.test.ts:219]: 用例有点“啰嗦”，用最小集的路由和 mock，能覆盖功能就够了。

Comment (ycjcl868) [reply packages/preset-built-in/src/plugins/commands/dev/mock/utils.test.ts:219]: ok

**Pull request merged**